src/remat.jl

```diff
@@ -550,31 +550,35 @@ function rowlengths(A::ReMat)
 end
 
 """
-    scaleinflate!(L::AbstractMatrix, Λ::ReMat)
+    copyscaleinflate!(L::AbstractMatrix, A::AbstractMatrix, Λ::ReMat)
 
-Overwrite L with `Λ'LΛ + I`
+Overwrite L with `Λ'AΛ + I`
 """
-function scaleinflate! end
+function copyscaleinflate! end
 
-function scaleinflate!(Ljj::Diagonal{T}, Λj::ReMat{T,1}) where {T}
-    Ljjd = Ljj.diag
-    broadcast!((x, λsqr) -> x * λsqr + 1, Ljjd, Ljjd, abs2(only(Λj.λ.data)))
+function copyscaleinflate!(Ljj::Diagonal{T}, Ajj::Diagonal{T}, Λj::ReMat{T,1}) where {T}
+    Ldiag, Adiag = Ljj.diag, Ajj.diag
+    broadcast!((x, λsqr) -> x * λsqr + one(T), Ldiag, Adiag, abs2(only(Λj.λ.data)))
     return Ljj
 end
 
-function scaleinflate!(Ljj::Matrix{T}, Λj::ReMat{T,1}) where {T}
+function copyscaleinflate!(Ljj::Matrix{T}, Ajj::Diagonal{T}, Λj::ReMat{T,1}) where {T}
+    fill!(Ljj, zero(T))
     lambsq = abs2(only(Λj.λ.data))
-    @inbounds for i in diagind(Ljj)
-        Ljj[i] *= lambsq
-        Ljj[i] += one(T)
+    @inbounds for (i, a) in enumerate(Ajj.diag)
```

Comment:
since we only ever create `Diagonal{T, Vector{T}}` this shouldn't cause trouble, but strictly speaking, there could be `Diagonal{T, StarWarsVector{T}}` which would cause undefined behavior with `@inbounds`

Reply:
Anyone who chooses to use a `StarWarsVector` in real life deserves undefined behavior. :smile: